handler/pair.go

```diff
@@ -0,0 +1,47 @@
+package handler
+
+import (
+	"fmt"
+	"net/http"
+	"path"
+
+	"github.com/gorilla/mux"
+
+	"github.com/cloudradar-monitoring/plexus/api"
+)
+
+// Pair godoc
+// @Summary Gets the pairing code info.
+// @Tags session
+// @Produce application/json
+// @Param code path string true "code"
+// @Security BasicAuth
+// @Success 200 {object} api.URLResponse
+// @Failure 401 {object} api.Error
+// @Failure 404 {object} api.Error
+// @Router /session/{code} [get]
+func (h *Handler) Pair(rw http.ResponseWriter, r *http.Request) {
+	h.codeLock.Lock()
+	id, ok := h.codes[mux.Vars(r)["id"]]
+	h.codeLock.Unlock()
```

Comment:
checkSessionAuthentication requires a lock of h.lock.
```suggestion
	h.lock.Lock()
	defer h.lock.Unlock()
	id, ok := h.codes[mux.Vars(r)["id"]]
```